addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,719 @@
+        export_string_translation=False,
+        copy=False,
+    )
+    l10n_vn_edi_invoice_symbol = fields.Many2one(
+        string='Invoice Symbol',
+        comodel_name='sinvoice.symbol',
+        compute='_compute_l10n_vn_edi_invoice_symbol',
+        readonly=False,
+        store=True,
+    )
+    l10n_vn_edi_invoice_number = fields.Char(
+        string='Invoice Number',
+        help='Invoice Number as appearing on SInvoice.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_reservation_code = fields.Char(
+        string='Secret Code',
+        help='Secret code that can be used by a customer to lookup an invoice on SInvoice.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_issue_date = fields.Datetime(
+        string='Issue Date',
+        help='Date of issue of the invoice on the e-invoicing system.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_sinvoice_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_file_id', 'l10n_vn_edi_sinvoice_file'),
+        depends=['l10n_vn_edi_sinvoice_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_file = fields.Binary(
+        string='SInvoice json File',
+        copy=False,
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_xml_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_xml_file_id', 'l10n_vn_edi_sinvoice_xml_file'),
+        depends=['l10n_vn_edi_sinvoice_xml_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_xml_file = fields.Binary(
+        string='SInvoice xml File',
+        copy=False,
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_pdf_file_id', 'l10n_vn_edi_sinvoice_pdf_file'),
+        depends=['l10n_vn_edi_sinvoice_pdf_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_pdf_file = fields.Binary(
+        string='SInvoice pdf File',
+        copy=False,
+        export_string_translation=False,
+    )
+    # Replacement/Adjustment fields
+    l10n_vn_edi_agreement_document_name = fields.Char(
+        string='Agreement Name',
+        copy=False,
+    )
+    l10n_vn_edi_agreement_document_date = fields.Datetime(
+        string='Agreement Date',
+        copy=False,
+    )
+    l10n_vn_edi_adjustment_type = fields.Selection(
+        string='Adjustment type',
+        selection=[
+            ('1', 'Money adjustment'),
+            ('2', 'Information adjustment'),
+        ],
+        copy=False,
+    )
+    # Only used in case of replacement invoice.
+    l10n_vn_edi_replacement_origin = fields.Many2one(
```

Comment:
Isn't it? If there is a flow where reversing (for modification) creates a single reverse entry for multiple invoices, it would be good to know as I would have to block it.
The API also does not support replacing multiple invoices with a single replacement one, it's always 1-1.